DAQ/src/StrawHitRecoFromFragments_module.cc

```diff
@@ -0,0 +1,360 @@
+//
+// This module transforms StrawDigi objects into StrawHit objects
+//
+// Original author David Brown, LBNL
+// Merged with flag and position creation B. Echenard, CalTech
+//
+// framework
+#include "art/Framework/Principal/Event.h"
+#include "art/Framework/Principal/Handle.h"
+#include "GeometryService/inc/GeomHandle.hh"
+#include "art/Framework/Core/EDProducer.h"
+#include "GeometryService/inc/DetectorSystem.hh"
+#include "art/Framework/Core/ModuleMacros.h"
+#include "art_root_io/TFileService.h"
+
+// conditions
+#include "ProditionsService/inc/ProditionsHandle.hh"
+#include "ConditionsService/inc/ConditionsHandle.hh"
+#include "ConditionsService/inc/AcceleratorParams.hh"
+#include "ConditionsBase/inc/TrackerCalibrationStructs.hh"
+#include "ConfigTools/inc/ConfigFileLookupPolicy.hh"
+#include "GeometryService/inc/GeomHandle.hh"
+#include "TrackerGeom/inc/Tracker.hh"
+#include "TrackerConditions/inc/StrawResponse.hh"
+#include "TrackerConditions/inc/TrackerStatus.hh"
+
+#include "TrkHitReco/inc/PeakFit.hh"
+#include "TrkHitReco/inc/PeakFitRoot.hh"
+#include "TrkHitReco/inc/PeakFitFunction.hh"
+#include "TrkHitReco/inc/ComboPeakFitRoot.hh"
+#include "TrkHitReco/inc/StrawHitRecoUtils.hh"
+
+#include "RecoDataProducts/inc/ProtonBunchTime.hh"
+#include "DataProducts/inc/StrawEnd.hh"
+#include "RecoDataProducts/inc/CaloCluster.hh"
+#include "RecoDataProducts/inc/StrawDigi.hh"
+#include "RecoDataProducts/inc/ComboHit.hh"
+#include "RecoDataProducts/inc/StrawHit.hh"
+
+#include "art/Framework/Principal/Handle.h"
+#include "mu2e-artdaq-core/Overlays/CalorimeterFragment.hh"
+#include "mu2e-artdaq-core/Overlays/FragmentType.hh"
+#include "mu2e-artdaq-core/Overlays/TrackerFragment.hh"
+#include <artdaq-core/Data/Fragment.hh>
+
+#include "TH1F.h"
+
+#include <memory>
+
+
+namespace art {
+class StrawHitRecoFromFragments;
+}
+
+class art::StrawHitRecoFromFragments : public art::EDProducer {
+  public:
+    using Name=fhicl::Name;
+    using Comment=fhicl::Comment;
+    struct Config {
+      fhicl::Atom<int> diag{ Name("diagLevel"), Comment("Diag level"), 0};
```

Comment:
can you please remove these defaults and move them in the corresponding module instance in the prolog file?